guida/src/app/api/posts/route.ts

```diff
@@ -75,18 +83,28 @@ export async function GET(request : Request) {
  */
 export async function PUT(request: Request) {
     try {
-        const { id, title, content, published } : PutRequestPost = await request.json(); // Parse the request body
-
-        // Update the post in the database
-        const updatedPost = await prisma.post.update({
-            where: { id: Number(id) }, // Update post by ID
-            data: { title, content, published } // The fields to update
-        });
-
-        return NextResponse.json({
-            message: `Successfully updated post with id: ${id}`,
-            post: updatedPost
-        }, {status: 200}); // Return the updated post
+        const session = await auth();
+        if (session) {
+            const { id, title, content, published, authorId } : SubmittablePost = await request.json();
```

Comment:
Perhaps get these attributes from post instead of having them hardcoded. If more attributes are added, we would need to edit the code in multiple places.

Reply:
in line 88, an interface is used. Therefore if we ever change what info we fetch, we need to change the interface and then change every instance where SubmittablePost is used so it will break everywhere else. We think it is reasonable to do it this way.